### ISSUE TITLE
iOS/Xamarin Forms crashes with 

### ISSUE THREAD

**Issue by dannyatger** [smartlook/smartlook-mobile-issue-tracker#145]
**Platform**

both

**Technology**

Xamarin.Forms 4.8.

**Build instructions [optional]**

**Rendering [optional]**

**Affected devices [optional]**

**Smartlook SDK version**

I tried Smartlook version 1.5.2 as well as 1.7.8 but both crashes.

**Summary**

I use our app as normal but after few seconds of interacting, it crashes.

It doesn't crash if I comment out Smartlook.Analytics.SetupAndStartRecording() on iOS.

(The same app on Android works OK with or without Smartlook).

**Steps to reproduce [optional]**

**Example Project [optional]**

**Relevant logs and/or screenshots**

```
2022-03-25 08:32:55.870 HCMobile.iOS[517:45833] 
2022-03-25 08:33:02.394 HCMobile.iOS[517:45833] Xamarin.iOS: Internal consistency error, please file a bug (https://github.com/xamarin/xamarin-macios/issues/new). Additional data: found managed object 0x11b802250=0x0 (Xamarin.Forms.Platform.iOS.FrameRenderer+FrameView) in native object 0x11aedf490 (NSKVONotifying_Xamarin_Forms_Platform_iOS_FrameRenderer_FrameView).
```
I've opened up a ticket with Xamarin.iOS so hopefully they can help me create a simple app to reproduce this problem.  But whatever you guys are doing behind the scenes, Xamarin or Xamarin.Forms is not liking it.

**Comment by dannyatger** [comment]
Here's a meaningful crash dump.  Does this help point towards a way around this crash?  (I tried closing session but that still made it crash).  (The iOS ticket is https://github.com/xamarin/xamarin-macios/issues/14508 )

Xamarin.iOS: Internal consistency error, please file a bug (https://github.com/xamarin/xamarin-macios/issues/new). Additional data: found managed object 0x11970b3f8=0x0 (Xamarin.Forms.Platform.iOS.FrameRenderer+FrameView) in native object 0x125e953f0 (NSKVONotifying_Xamarin_Forms_Platform_iOS_FrameRenderer_FrameView).

=================================================================
	Native Crash Reporting
=================================================================
Got a abrt while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x104dbe4d8 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x104db4624 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x104dbda20 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x1f1a47c10 - /usr/lib/system/libsystem_platform.dylib : <redacted>
	0x1f1a55378 - /usr/lib/system/libsystem_pthread.dylib : pthread_kill
	0x18b59cf50 - /usr/lib/system/libsystem_c.dylib : abort
	0x104f758e0 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : xamarin_get_block_descriptor
	0x104f77b94 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : xamarin_get_block_descriptor
	0x10065ed50 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x100686564 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x182cd88e8 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x104f7d8e4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x104f7da88 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x10065f2b4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x198f0f4bc - /usr/lib/libobjc.A.dylib : <redacted>
	0x198f0b888 - /usr/lib/libobjc.A.dylib : objc_autoreleasePoolPop
	0x182cd8ac4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x104f7d8e4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x104f7da88 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x10065f2b4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x198f0f4bc - /usr/lib/libobjc.A.dylib : <redacted>
	0x198f0b888 - /usr/lib/libobjc.A.dylib : objc_autoreleasePoolPop
	0x182cd8ac4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x104f7d8e4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x104f7da88 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x10065f2b4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x198f0f4bc - /usr/lib/libobjc.A.dylib : <redacted>
	0x198f0b888 - /usr/lib/libobjc.A.dylib : objc_autoreleasePoolPop
	0x182cd8ac4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x104f7d8e4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x104f7da88 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x10065f2b4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x198f0f4bc - /usr/lib/libobjc.A.dylib : <redacted>
	0x198f0b888 - /usr/lib/libobjc.A.dylib : objc_autoreleasePoolPop
	0x182cd8ac4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x182d539c4 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x104f7d8e4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x104f7da88 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : _Z9__isctypeim
	0x1006627e4 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x180646a00 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x18065a6ec - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x198f0f4bc - /usr/lib/libobjc.A.dylib : <redacted>
	0x198f0b888 - /usr/lib/libobjc.A.dylib : objc_autoreleasePoolPop
	0x18064c64c - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : _CFAutoreleasePoolPop
	0x181e9a47c - /System/Library/Frameworks/Foundation.framework/Foundation : <redacted>
	0x1806f30d0 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x180703d90 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x18063e098 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x1806438a4 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : <redacted>
	0x180657468 - /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation : CFRunLoopRunSpecific
	0x19c1f738c - /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices : GSEventRunModal
	0x182ffa5d0 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : <redacted>
	0x182d78f74 - /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore : UIApplicationMain
	0x101a2fb58 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x1019863d0 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x1006aa1d0 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x100ab7df0 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x104dc5d98 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : mono_pmip
	0x104e7f9c8 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : mono_pmip
	0x104e85cd0 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : mono_pmip
	0x104da4184 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x104f8b7b0 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : xamarin_log
	0x1006a9e40 - /private/var/containers/Bundle/Application/01FF1FA4-84FB-4059-942E-6BCFC6D40E5E/HCMobile.iOS.app/HCMobile.iOS : sqlite3_sourceid
	0x1058e5aa4 - Unknown

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0x1b7d8f964):0x1b7d8f954  ff 0f 5f d6 c0 03 5f d6 10 29 80 d2 01 10 00 d4  .._..._..)......
0x1b7d8f964  03 01 00 54 7f 23 03 d5 fd 7b bf a9 fd 03 00 91  ...T.#...{......
0x1b7d8f974  34 ee ff 97 bf 03 00 91 fd 7b c1 a8 ff 0f 5f d6  4........{...._.
0x1b7d8f984  c0 03 5f d6 7f 23 03 d5 fd 7b bf a9 fd 03 00 91  .._..#...{......

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at UIKit.UIApplication:UIApplicationMain <0x00007>
	  at UIKit.UIApplication:Main <0x0005f>
	  at HCMobile.iOS.Application:Main <0x0007f>
	  at System.Object:runtime_invoke_dynamic <0x0010f>
=================================================================

**Comment by pavel-kroh-smartlook** [comment]
Hi @dannyatger , 

we know about the Xamarin.Forms related issues and unfortunately, we cannot fix them. 

Smartlook uses a low-level technology known as _swizzling_, that, if not done carefully by all involved parties, causes this kind of crashes. Xamarin.Forms is the only framework that causes this kind of issues when used along with Smartlook (standard Xamarin works fine). We tried to debug if to find a work around, but after some effort (and support of another client of us interested to make it work) we decided not to fix it and not support Xamarin.Forms apps.
